### ISSUE TITLE
Doesn't relaunch - ask me to terminate batch job 3 times

### ISSUE THREAD

**Issue by jonyeezs** [dblock/guard-rack#22]
Running on windows 10.

This is my gemfile:

```
source 'https://rubygems.org'
ruby '2.1.7'

gem 'sinatra', require: 'sinatra/base'
gem 'thin'
gem 'activesupport', '~> 4.2', '>= 4.2.5', require: 'active_support/all'
gem 'json'
gem 'google_drive'

group :development do
  gem 'sinatra-contrib'
  gem 'win32console'
  gem 'wdm', '>= 0.1.0' if Gem.win_platform?
  gem 'guard'
  gem 'guard-rack'
  # gem 'rerun' works only on Unix
end
```

My GuardFile:

```
guard 'rack', port: 9292 do
  watch('Gemfile.lock')
  watch('config.ru')
  watch('server.rb')
  watch(%r{^config/.*})
  watch(%r{^app/(?!routes)([a-zA-Z]+)/.*}) # routes will be watched by sinatra::reloader
end
```

When i change a file i get this error:

```
Thin web server (v1.6.4 codename Gob Bluth)
Maximum connections set to 1024
Listening on 0.0.0.0:9292, CTRL+C to stop
127.0.0.1 - - [02/Apr/2016:00:09:00 +1000] "GET /budgets/year HTTP/1.1" 200 87 0.1421

00:09:29 - INFO - Restarting Rack...

^C^C^CTerminate batch job (Y/N)? Terminate batch job (Y/N)? Terminate batch job (Y/N)?
```

**Comment by dblock** [comment]
Oy, Windows :(

I am afraid can't be of much help from the top of my head, but clearly guard things there were changes and the process aborts?

**Comment by jonyeezs** [comment]
@dblock there were changes made. So Guard was working. But stopping rack was the issue.
I think it tried to do a ctrl+C and the "Terminate batch job" message showed up and it couldn't exit from that.

Maybe I'll have to find a way to bypass that blocker

**Comment by jonyeezs** [comment]
So what seems to be the problem is rackup executes a batch file.
And windows prompts

> Terminate batch job (Y/N)?

on ctrl+C. Guard doesn't seem to work well when that appears...

Apparently you can bypass the batch if you run the ruby code directly by
`ruby -S rackup`

I tried doing this manually. And it works. But if i put it in the `cmd:`, nothing happens.

No idea how spawnp works...really.

**Comment by dblock** [comment]
If there's something we can do here to make it work on Windows I'll gladly take a pull request.

**Comment by jonyeezs** [comment]
I've been trying a few things but no success. :(

**Comment by dblock** [comment]
Closed via #23.
